src/transformers/models/beit/configuration_beit.py

```diff
@@ -183,6 +200,26 @@ def __init__(
         self.auxiliary_concat_input = auxiliary_concat_input
         self.semantic_loss_ignore_index = semantic_loss_ignore_index
 
+        # handle backwards compatibility
+        if "segmentation_indices" in kwargs:
+            logger.warning(
+                "The `segmentation_indices` argument is deprecated and will be removed in a future version, use `out_indices` instead.",
+                FutureWarning,
+            )
+            out_indices = kwargs.pop("segmentation_indices")
+        elif out_indices == [3, 5, 7, 11]:
+            out_features = ["stage4", "stage6", "stage8", "stage12"]
+        elif out_indices == [7, 11, 15, 23]:
+            out_features = ["stage8", "stage12", "stage16", "stage24"]
```

Comment:
This hard coding isn't needed. Given stage_names and out_features you can directly get out_indices i.e.:

```suggestion
out_features, out_indices = get_aligned_output_features_output_indices(
    out_features=None, out_indices=out_indices, stage_names=stage_names
)
```